src/Compilers/CSharp/Portable/CodeGen/EmitStackAllocInitializer.cs

```diff
@@ -45,7 +45,11 @@ private void EmitStackAlloc(TypeSymbol type, BoundArrayInitialization? inits, Bo
                 var sizeInBytes = elementType.EnumUnderlyingTypeOrSelf().SpecialType.SizeInBytes();
 
                 ImmutableArray<byte> data = GetRawData(initExprs);
+#if NET8_0_OR_GREATER
+                if (!data.AsSpan().ContainsAnyExcept(data[0]))
```

Comment:
>if (!data.AsSpan().ContainsAnyExcept(data[0])) [](http://example.com/codeflow?start=16&length=46)

We can add an overload of ```All``` which takes an extra argument and passes it to the predicate. I think that would be preferred over a conditional compilation in this case.

Reply:
In fact we already have the overload in ImmutableArrayExtensions:
```
public static bool All<T, TArg>(this ImmutableArray<T> array, Func<T, TArg, bool> predicate, TArg arg)
```